lib/attr_extras/attr_initialize/attributes.rb

```diff
@@ -0,0 +1,36 @@
+module AttrExtras
+  class AttrInitialize
+    class Attributes
+      def initialize(names_with_array_of_hash_arguments)
+        @names_with_array_of_hash_arguments = names_with_array_of_hash_arguments
+      end
+
+      def positional
+        @attributes_names ||= names_with_array_of_hash_arguments.take_while { |name_or_array| !name_or_array.is_a?(Array) }
+      end
```

Comment:
Hm.

This makes a (reasonable) assumption that we never have positional arguments after the hash arguments, but my local attr_extras seems to allow that:

``` ruby
class Snake; pattr_initialize :foo, [:bar], :baz; end
Snake.new(1, { bar: 2 }, 3).send(:bar)  # => 2
```

Do you think it would be an unproportional bother to raise if that happens? I would prefer the lib to explode instead of silently doing weird things.